net/dns/wsl_windows.go

```diff
@@ -9,19 +9,20 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"os/user"
 	"strings"
 	"syscall"
 	"unicode/utf16"
 
+	"golang.org/x/sys/windows"
 	"tailscale.com/types/logger"
 )
 
 // wslDistros reports the names of the installed WSL2 linux distributions.
 func wslDistros(logf logger.Logf) []string {
-	cmd := exec.Command("wsl.exe", "-l")
-	cmd.SysProcAttr = &syscall.SysProcAttr{HideWindow: true}
-	b, err := cmd.CombinedOutput()
+	b, err := wslCombinedOutput(exec.Command("wsl.exe", "-l"))
```

Comment:
I don't think this is going to work in practice: the machine first boots up and nobody's logged in yet, so WTSGetActiveConsoleSessionId will return an error, and then dns.NewOSConfigurator will never run again.